packages/pymedphys_dicom/src/pymedphys_dicom/ct/extend.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (C) 2019 Cancer Care Associates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version (the "AGPL-3.0+").
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+
+import os
+from copy import deepcopy
+from glob import glob
+import datetime
+import random
+from collections import deque
+
+import pydicom
+from ..constants.uuid import PYMEDPHYS_ROOT_UID
+
+
+def extend(input_dir, output_dir, index_to_copy, number_of_slices):
+    filepaths = glob(os.path.join(input_dir, '*'))
+    dicom_datasets = load_dicom_into_deque(filepaths)
```

Comment:
`dicom` in `dicom_datasets` is superfluous. We're in the `dicom` package...

Reply:
Maybe when it's on a DICOM object...

I'd agree if I called teh function `dicom_ct_extend` which would be superfluous. But the variable name doesn't necessarily have to be DICOM...

Reply:
The `dicom` module is pretty analogous to a dicom class. And given our extension use of `dataset`/`ds`, dataset is effectively our "object" of that class (for now)...

I see the value of "explicit is better than implicit", so if you're firm on this, I'll let it go.

I guess I'm just starting to cringe at our growing lack of consistency in our variable names for instances (or collections of instances) of `pydicom`'s `Dataset` . We now have lots of `dicom_dataset`, `dataset` AND `ds`.

Reply:
Yeah... I see the cringe. ... In this case a believe it helps the readability of this code... Okay we stick with this here?

Reply:
Yep :)

Reply:
Kk :) thanks :)